test/unit/module/core/count_true.cpp

```diff
@@ -9,85 +9,124 @@
+TTS_CASE_TPL("Check eve::count_true return type (scalar)", eve::test::scalar::all_types)
 <typename T>(tts::type<T>)
 {
-  eve::logical<T> data(true);
-  constexpr auto  cardinal = eve::cardinal_v<T>;
+  TTS_EXPR_IS((eve::count_true(eve::logical<T>())), std::ptrdiff_t);
+  TTS_EXPR_IS((eve::count_true[eve::ignore_none](eve::logical<T>())), std::ptrdiff_t);
+  TTS_EXPR_IS((eve::count_true[true](eve::logical<T>())), std::ptrdiff_t);
+};
 
-  TTS_EQUAL(eve::count_true[eve::ignore_none](data), cardinal);
-  TTS_EQUAL(eve::count_true[eve::ignore_all](data), 0);
-  TTS_EQUAL(eve::count_true[eve::ignore_first(cardinal - 1)](data), 1);
-  TTS_EQUAL(eve::count_true[eve::keep_first(cardinal - 1)](data), cardinal - 1);
-  TTS_EQUAL(eve::count_true[eve::ignore_last(cardinal - 1)](data), 1);
-  TTS_EQUAL(eve::count_true[eve::keep_last(cardinal - 1)](data), cardinal - 1);
+TTS_CASE("Check eve::count_true return type (edge cases)")
+{
+  TTS_EXPR_IS((eve::count_true(bool{})), std::ptrdiff_t);
+  TTS_EXPR_IS((eve::count_true[eve::ignore_none](bool{})), std::ptrdiff_t);
+  TTS_EXPR_IS((eve::count_true[true](bool{})), std::ptrdiff_t);
+};
+
+template<typename T>
+bool mask_at(T mask, int idx)
+{
+  if constexpr (std::same_as<T, bool>) return mask;
+  else                                 return mask.get(idx);
+}
 
-  if constexpr( cardinal >= 2 )
+template<typename T, typename U>
+std::ptrdiff_t manual_count_true(T v, U mask)
+{
+  if constexpr (eve::conditional_expr<U>)
+  {
+    return manual_count_true(v, expand_mask(mask, eve::as(v)));
+  }
+  else if constexpr (eve::scalar_value<T> || std::same_as<T, bool>)
   {
-    TTS_EQUAL(eve::count_true[eve::ignore_first(1) && eve::ignore_last(1)](data), cardinal - 2);
+    return std::ptrdiff_t{v && mask};
   }
   else
   {
-    //   TTS_EQUAL( eve::count_true[eve::ignore_first(1) && eve::ignore_last(1)](data), 0);
+    std::ptrdiff_t expected = 0;
+
+    for (int i = 0; i < v.size(); ++i)
+        if (v.get(i) && mask_at(mask, i))
+          ++expected;
+
+    return expected;
   }
+}
+
+template<typename T>
+void test_count_true(T v)
+{
+  TTS_EQUAL(eve::count_true(v), manual_count_true(v, true));
+  TTS_EQUAL(eve::count_true[true](v), manual_count_true(v, true));
+  TTS_EQUAL(eve::count_true[false](v), 0);
+  TTS_EQUAL(eve::count_true[eve::ignore_none](v), manual_count_true(v, true));
+
+  if constexpr (eve::simd_value<T>)
+  {
+    T m = [](auto i, auto) { return i % 2 == 0; };
+    TTS_EQUAL(eve::count_true[m](v), manual_count_true(v, m));
+
+    TTS_EQUAL(eve::count_true[eve::ignore_all](v), 0);
+    TTS_EQUAL(eve::count_true[eve::keep_first(1)](v), manual_count_true(v, eve::keep_first(1)));
+    TTS_EQUAL(eve::count_true[eve::ignore_first(1)](v), manual_count_true(v, eve::ignore_first(1)));
+    TTS_EQUAL(eve::count_true[eve::keep_last(1)](v), manual_count_true(v, eve::keep_last(1)));
+    TTS_EQUAL(eve::count_true[eve::ignore_last(1)](v), manual_count_true(v, eve::ignore_last(1)));
+  }
```

Comment:
The covergae for ignore is inusfficient.

Also - why are you copy pasting here?

Call

template<typename T>
void test_count_true(T v, ignore) or smth